### ISSUE TITLE
Addon v2 support?

### ISSUE THREAD

**Issue by runspired** [empress/field-guide#64]

**Comment by mansona** [comment]
Yes it is a matter of tinkering and the good news is that I have a plan! The bad news is that it's a little bit far off right now. 

The first issue we have is that there isn't much consensus yet on what a V2 addon looks like. Sure the functional parts are being finalised but the design of the actual official blueprint is still very much in the air, so any work that we do in field-guide to "just work out of the box" will start feeling like trying to hit a moving target very quickly. We will likely have to wait for the inevitable RFC for the default addon blueprint before we will know what this looks like completely.

The good news is that this V2 addon structure doesn't block us from doing **something** that might work with a V2 addon. I'm currently heavily focusing on https://github.com/empress/field-guide/pull/58 to make sure that people's ember-try scenarios can be all-green with field-guide (which is in turn blocked by https://github.com/ef4/ember-auto-import/pull/528). Once this bit of work is done I'm going to start expanding on an idea that [I presented briefly in my EmberFest talk last year](https://www.youtube.com/watch?v=vmspW-tqV8A), [Bottled Ember](https://github.com/empress/bottled-ember)

The idea of Bottled Ember gets around the issue that V2 addons will need to use workspaces or a monorepo to have a "test app" and a "documentation app" by providing a full Ember app in a single npm package that can be used to build the host app for either tests or in our case the host app for the field-guide documentation. This is obviously a bit of work to get right but I am dedicated to making it work and ergonomic for people, it just might take me a little time ;) 

I hope that gives you a bit of insight into what I'm working on to bridge this gap 👍
